rust_snuba/rust_arroyo/src/processing/dlq.rs

```diff
@@ -0,0 +1,95 @@
+#![allow(dead_code)]
+use crate::types::{BrokerMessage, Partition};
+use std::cmp::Ordering;
+use std::collections::{BTreeMap, VecDeque};
+
+/// Stores messages that are pending commit. This is used to retreive raw messages
+// in case they need to be placed in the DLQ.
+pub struct BufferedMessages<TPayload> {
+    buffered_messages: BTreeMap<Partition, VecDeque<BrokerMessage<TPayload>>>,
+}
+
+impl<TPayload> BufferedMessages<TPayload> {
+    pub fn new() -> Self {
+        BufferedMessages {
+            buffered_messages: BTreeMap::new(),
+        }
+    }
+
+    // Add message to the buffer.
+    pub fn append(&mut self, message: BrokerMessage<TPayload>) {
+        match self.buffered_messages.get_mut(&message.partition) {
+            Some(messages) => {
+                messages.push_back(message);
+            }
+            None => {
+                self.buffered_messages
+                    .insert(message.partition.clone(), VecDeque::from([message]));
+            }
+        };
+    }
+
+    // Return the message at the given offset or None if it is not found in the buffer.
+    // Messages up to the offset for the given partition are removed.
+    pub fn pop(&mut self, partition: &Partition, offset: u64) -> Option<BrokerMessage<TPayload>> {
+        if let Some(messages) = self.buffered_messages.get_mut(partition) {
+            #[allow(clippy::never_loop)]
```

Comment:
are you sure this lint is alerting on the code? can't see why it would, is it possible it alerted on an earlier version of it?

Reply:
yeah.. i've tried everything but it is alerting